docs/src/main/asciidoc/amazon-snapstart.adoc

```diff
@@ -0,0 +1,184 @@
+
+The classes to preload are listed in two places:
+
+1. extensions can produce a list of classes (using the `io.quarkus.deployment.builditem.PreloadClassBuildItem` build item)
+2. you can add a `src/main/resources/META-INF/quarkus-preload-classes.txt` file listing the classes to preload, such as:
+
+[source]
+----
+com.amazonaws.services.lambda.runtime.LambdaRuntimeInternal
+com.fasterxml.jackson.annotation.JsonAlias
+com.fasterxml.jackson.annotation.JsonFormat$Feature
+com.fasterxml.jackson.core.exc.InputCoercionException
+com.fasterxml.jackson.core.exc.StreamWriteException
+com.fasterxml.jackson.core.io.ContentReference
+com.fasterxml.jackson.core.io.IOContext
+com.fasterxml.jackson.core.io.JsonEOFException
+com.fasterxml.jackson.core.io.MergedStream
+com.fasterxml.jackson.core.io.NumberInput
+com.fasterxml.jackson.core.io.NumberOutput
+com.fasterxml.jackson.core.io.UTF32Reader
+com.fasterxml.jackson.core.json.ByteSourceJsonBootstrapper
+com.fasterxml.jackson.core.json.JsonReadContext
+com.fasterxml.jackson.core.json.JsonWriteContext
+com.fasterxml.jackson.core.json.UTF8StreamJsonParser
+com.fasterxml.jackson.core.JsonEncoding
+com.fasterxml.jackson.core.JsonGenerationException
+com.fasterxml.jackson.core.JsonLocation
+com.fasterxml.jackson.core.JsonStreamContext
+com.fasterxml.jackson.core.JsonToken
+...
+----
+
+The format is simple: one class per line.
+
+=== Computing the class list
+
+NOTE: That step is particularly not user-friendly. We plan to improve it.
+
+To compute the list of classes, we recommend deploying your function and setting the `JAVA_TOOL_OPTIONS` environment variable to `-verbose:class`.
+Then execute your function and retrieve the log (in CloudWatch).
+You should be able to extract the class names using sed/awk or any text editor.
+
+=== Application class list
+
+By default, Quarkus generates the class list of the classes included in your application (including the classes generated by Quarkus).
+So, you do not have to repeat them in the `quarkus-preload-classes.txt` file.
+
+You can disable this feature using:
+
+[source, properties]
+----
+quarkus.snap-start.generate-application-class-list=false
+----
+
+=== Disable preloading
+
+You can disable the preloading of classes using:
+
+[source, properties]
+----
+quarkus.snap-start.preload-classes=false
+----
+
+=== Skipping class initialization
+
+By default, the classes are preloaded, the classes are also _initialized_, meaning it also resolves the dependent classes.
+You can disable this behavior using:
+
+[source, properties]
+----
+quarkus.snap-start.initialize-classes=false
+----
+
+== Client Priming
+
+Client _priming_ is a technic that allows initializing a client during the snapshotting process, so, it's already fully functional during the application runtime.
+
+There are two ways to achieving _priming_:
+
+1. initialize the client in a `static` block, which, thanks to class preloading will be executed before the snapshot
```

Comment:
should we warn or be explicit that if one does too much here your app might not build with native?

Reply:
Oh yeah! I forgot to mention that!